index.js

```diff
@@ -54,18 +68,204 @@ client.on("messageReactionAdd", async (reaction, user) => {
-        emojiReactToSend(reaction.message, user, firstTargetChannelId);
+    if (reaction.emoji.name === PIN_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, PIN_TARGET_CHANNEL_ID);
     }
 
     // Check if reaction emoji matches second target emoji:
-    if (reaction.emoji.name === secondTargetEmoji) {
-        emojiReactToSend(reaction.message, user, secondTargetChannelId);
+    if (reaction.emoji.name === UNREAD_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, UNREAD_TARGET_CHANNEL_ID);
     }
 
     // Check if reaction emoji matches third target emoji:
-    if (reaction.emoji.name === thirdTargetEmoji) {
-        emojiReactToSend(reaction.message, user, thirdTargetChannelId);
+    if (reaction.emoji.name === TODO_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, TODO_TARGET_CHANNEL_ID);
+    }
+
+    // Check emojis for deleting todo:
+    const message = reaction.message.content;
+
+    if (reaction.emoji.name === DELETE_TODO_EMOJI) {
+        if (message.match(USERNAME_REGEX)[0] === `<@${user.id}>`) {
+            deleteTodo[0] = true;
+        }
+        if (message.match(USERNAME_REGEX)[1] === `<@${user.id}>`) {
+            deleteTodo[1] = true;
+        }
+        if (deleteTodo[0] === true && deleteTodo[1] === true) {
+            reaction.message.delete();
+            deleteTodo[0] = false;
+            deleteTodo[1] = false;
+        }
+    }
+});
+
+// Event listener for reactions removed from messages:
+client.on("messageReactionRemove", async (reaction, user) => {
+    // Ignore bot reactions:
+    if (user.bot) {
+        return;
+    }
+
+    // Fetch partials if necessary:
+    if (reaction.partial) {
+        try {
+            await reaction.fetch();
+        }
+        catch (error) {
+            console.error("Something went wrong when fetching the reaction: ", error);
+            return;
+        }
+    }
+
+    // Check emojis removed for deleting todo:
+    const message = reaction.message.content;
+
+    if (message.match(USERNAME_REGEX)[0] === `<@${user.id}>`) {
+        deleteTodo[0] = false;
+    }
+    if (message.match(USERNAME_REGEX)[1] === `<@${user.id}>`) {
+        deleteTodo[1] = false;
+    }
+});
+
+// Event listener for messages sent:
+client.on("messageCreate", async (message) => {
+    // Skip bot's own messages:
+    if (message.author.bot) {
+        return;
+    }
+
+
+    // Check if message is a reply:
+    if (message.reference) {
+        const originalMessageId = message.reference.messageId;
+
+        try {
+            // Fetch original message being replied to:
+            const originalMessage = await message.channel.messages.fetch(originalMessageId);
+            const originalMessageLink = `https://discord.com/channels/${originalMessage.guildId}/${originalMessage.channelId}/${originalMessage.id}`;
+
+            // Add todo:
+            if (message.content.substring(0, 5).match(TODO_REGEX)) {
```

Comment:
Can you explain why you're only checking the first 5 characters here? Could add a comment here explaining why this is necessary

Reply:
This is for the case where the message is only "TODO" as a reply, so only the first few characters need to be checked. But there's probably a better way? Will add comment nonetheless